securedrop/store.py

```diff
@@ -68,17 +68,26 @@ def path(*s):
     return absolute
 
 
-def get_bulk_archive(filenames, zip_directory=''):
+def get_bulk_archive(items_selected, zip_directory=''):
     zip_file = tempfile.NamedTemporaryFile(prefix='tmp_securedrop_bulk_dl_',
                                            dir=config.TEMP_DIR,
                                            delete=False)
+    sources = set([i.source.journalist_designation for i in items_selected])
+    # The below nested for-loops are there to create a more usable
+    # folder structure per #383
     with zipfile.ZipFile(zip_file, 'w') as zip:
-        for filename in filenames:
-            verify(filename)
-            zip.write(filename, arcname=os.path.join(
-                zip_directory,
-                os.path.basename(filename)
-            ))
+        for source in sources:
+            submissions = [s for s in items_selected if s.source.journalist_designation == source]
+            for index, submission in enumerate(submissions, start=1):
+                filename = path(submission.source.filesystem_id,
+                                submission.filename)
+                verify(filename)
+                zip.write(filename, arcname=os.path.join(
+                    zip_directory,
+                    source,
+                    "%s_%s" % (index, submission.source.last_updated.date()),
```

Comment:
This is not the index we want here. We want `submission.filename.split('_')[0]`.